src/main/java/com/google/sps/data/TimeRange.java

```diff
@@ -21,10 +21,40 @@
  * providing methods to make ranges easier to work with (e.g. {@code overlaps}).
  */
 public final class TimeRange {
-  public static final int START_OF_DAY = getTimeInMinutes(0, 0);
-  public static final int END_OF_DAY = getTimeInMinutes(23, 59);
+  public static final int SUNDAY = 0;
+  public static final int MONDAY = 1;
+  public static final int TUESDAY = 2;
+  public static final int WEDNESDAY = 3;
+  public static final int THURSDAY = 4;
+  public static final int FRIDAY = 5;
+  public static final int SATURDAY = 6;
 
-  public static final TimeRange WHOLE_DAY = new TimeRange(0, 24 * 60);
+  public static final int START_OF_SUNDAY = getTimeInMinutes(SUNDAY, 0, 0);
+  public static final int END_OF_SUNDAY = getTimeInMinutes(SUNDAY, 23, 59);
+
+  public static final int START_OF_MONDAY = getTimeInMinutes(MONDAY, 0, 0);
+  public static final int END_OF_MONDAY = getTimeInMinutes(MONDAY, 23, 59);
+
+  public static final int START_OF_TUESDAY = getTimeInMinutes(TUESDAY, 0, 0);
+  public static final int END_OF_TUESDAY = getTimeInMinutes(TUESDAY, 23, 59);
+
+  public static final int START_OF_WEDNESDAY = getTimeInMinutes(WEDNESDAY, 0, 0);
+  public static final int END_OF_WEDNESDAY = getTimeInMinutes(WEDNESDAY, 23, 59);
+
+  public static final int START_OF_THURSDAY = getTimeInMinutes(THURSDAY, 0, 0);
+  public static final int END_OF_THURSDAY = getTimeInMinutes(THURSDAY, 23, 59);
+
+  public static final int START_OF_FRIDAY = getTimeInMinutes(FRIDAY, 0, 0);
+  public static final int END_OF_FRIDAY = getTimeInMinutes(FRIDAY, 23, 59);
+
+  public static final int START_OF_SATURDAY = getTimeInMinutes(SATURDAY, 0, 0);
+  public static final int END_OF_SATURDAY = getTimeInMinutes(SATURDAY, 23, 59);
+
+  public static final int START_OF_WEEK = getTimeInMinutes(0, 0);
+  public static final int END_OF_WEEK = getTimeInMinutes(6, 23, 59);
```

Comment:
I wager you could greatly reduce the need for these `START_OF.../END_OF...` constants by updating the `fromStartDuration` method on this class to better match your use case.

Something like..

```
public static TimeRange fromStartDuration(int day, int hour, int minute, int durationMinutes) { ... }
```

Same for `fromStartEnd`. It'll likely make your life easier.